src/Hcal/HcalDigiProducer.cxx

```diff
@@ -95,27 +145,224 @@ void HcalDigiProducer::produce(framework::Event& event) {
+#ifdef NEWCODE
+
+    //******************//
+    //***  new code  ***//
+    //******************//
+
+    ldmx::HcalID detID(simHit.getID());
+    int section = detID.section();
+    int layer = detID.layer();
+    int strip = detID.strip();
+    ldmx::HcalDigiID posEndID(section, layer, strip, 0);
+    ldmx::HcalDigiID negEndID(section, layer, strip, 1);
+
+    //TODO: temporary implementation for getting the scintillator length
+    int length=(int)(2.0*hcalGeometry.getHalfTotalWidth(section,(layer%2==1?layer+1:layer-1)) + 0.5); //rounded to closest int to avoid mismatches due to precission of floating point numbers
+    int reflectorType=(section==ldmx::HcalID::HcalSection::BACK?0:2);
+    auto photonGenerator = photonGenerators_.find(std::pair<int,int>(length, reflectorType));
+    if(photonGenerator==photonGenerators_.end()) throw std::runtime_error("HcalDigiProducer::produce: Found an a scintillator for which we don't have a lookup table");
+
+    //randomly set a scintillation yield around the mean scintillation yield for each scintillator
+    //needs to be done here, because the geometry is not available in configure()
+    auto currentScintillationYield = scintillationYieldsAdjusted_.find(detID);
+    if(currentScintillationYield==scintillationYieldsAdjusted_.end())
+    {
+      double adjustedYield=0;
+      do
+      {
+        adjustedYield=randGaussQ_.fire(scintillationYield_, scintillationYield_*scintillationYieldSigma_);
+      } while(adjustedYield<scintillationYield_*scintillationYieldCutoffLow_ ||
+              adjustedYield>scintillationYield_*scintillationYieldCutoffHigh_);
+
+      currentScintillationYield = scintillationYieldsAdjusted_.emplace(detID,adjustedYield).first;
+    }
+    photonGenerator->second->SetScintillationYield(currentScintillationYield->second);
+
+    CLHEP::Hep3Vector pos1Local, pos2Local;
+    for(int i=0; i<3; ++i)
+    {
+      pos1Local[i]=simHit.getPreStepPosition().at(i);
+      pos2Local[i]=simHit.getPostStepPosition().at(i);
+  if(std::fabs(pos1Local[0])>10.0) throw std::runtime_error("step point bigger than lookup table thickness");
+  if(std::fabs(pos2Local[0])>10.0) throw std::runtime_error("step point bigger than lookup table thickness");
+  if(std::fabs(pos1Local[1])>25.0) throw std::runtime_error("step point bigger than lookup table width");
+  if(std::fabs(pos2Local[1])>25.0) throw std::runtime_error("step point bigger than lookup table width");
+  if(std::fabs(pos1Local[2])>length/2.0) throw std::runtime_error("step point bigger than lookup table length");
+  if(std::fabs(pos2Local[2])>length/2.0) throw std::runtime_error("step point bigger than lookup table length");
+    }
+
+    int nContributions = simHit.getNumberOfContribs();
+    if(nContributions==0) continue;
+    int trackID = simHit.getContrib(0).trackID;  //There should only be 1 contributions for Hcal hits
+    double charge = particleMap[trackID].getCharge();
+
+    int absorber=0;  //no absorber at back section
```

Comment:
What does "no absorber" at the back section mean?

Reply:
The counters at the back section of the Hcal have SiPMs on both ends, i.e. no absorbers. I use this as an initial value. The next couple of lines deal with all other Hcal sections and set the value of the absorber variable to 2 or -2, if necessary (the negative sign indicates that the absorber is at the negative end of the counter).